app/components/SingleProduct.js

```diff
@@ -48,64 +51,90 @@ class SingleProduct extends Component {
           color="primary"
+          style={{ marginBottom:"10px"}}
           onClick={() => addCartItem(cart.id, this.props.match.params.id, this.state.quantity)}
           disabled={selectedProduct.quantity === 0}
         >
+        <AddShoppingCartIcon fontSize={"small"} style={{ paddingRight: "10px" }} />
           Add to cart
         </Button>
+        </Grid>
         {
           user && user.isAdmin &&
           <Button
             variant="contained"
             color="primary"
+            style={{ marginBottom: "10px" }}
             onClick={() => this.props.history.push(`/products/edit/${selectedProduct.id}`)}
           >
             Edit Product
             </Button>
         }
-        <Button variant="contained" color="secondary"
-          onClick={() => {
-            clearSelectedProduct()
-            this.props.history.push('/')
-          }}
-        >Return to products
-          </Button>
-        <h3>
-          Average Rating: </h3>
-        {selectedProduct.numRatings === 0 ? <span style={{ fontStyle: 'italic' }}>No ratings</span> :
-          <Rating name="rating" value={Math.ceil(selectedProduct.averageRating)} readOnly size="small" />
-        }
+        
+          <Button>
+          <Link
+            to='/'
+            onClick={() => clearSelectedProduct()}
+          >Return to products</Link></Button>
+          </Grid>
+      
+            <Grid item sm={6}>
+              <Button color="secondary">
+                <Link
+                  to='/review'
+                > Leave Review</Link> </Button>
+                </Grid>
+         <Grid item sm={6}>
+              <h3>
+                Average Rating: </h3>
+              {selectedProduct.numRatings === 0 ? <span style={{ fontStyle: 'italic' }}>No ratings</span> :
+                <Rating name="rating" value={Math.ceil(selectedProduct.averageRating)} readOnly size="small" />
+              }
         <h3>
           Customer Reviews:
         </h3>
-        {reviews.length === 0 ? <span style={{ fontStyle: 'italic' }}>No reviews</span> :
-          <ul>
-            {reviews && reviews.map(review => {
-              return (
-                <li key={review.id} style={{ fontWeight: 'bold' }}>
-                  {review.user.name}: {review.title}
-                  <div>
-                    <div style={{ fontStyle: 'italic', fontWeight: 'normal', padding: '3px' }}>"{review.body}"</div>
-                  </div>
-                </li>
-              )
-            })}
-          </ul>
-        }
-      </div>
+        <div>
+          {reviews && reviews.map(review => {
+            return (
+              <React.Fragment key={review.id}>
+
+                {review.title}
+                <ul key>
+                  <li>{review.body}</li>
+                </ul>
+              </React.Fragment>
+            )
+          })}
+        </div>
+            </Grid>
+        </Grid>
+        </Container>
+      </React.Fragment>
```

Comment:
Can we keep some of the latest review/rating updates, like displaying 'no reviews' if there are none, etc